packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,360 @@
+if (process.argv.length != 3 && process.argv.length != 4) {
+  console.error(
+    'Usage: ' +
+      process.argv[0] +
+      ' ' +
+      process.argv[1] +
+      ' path/to/rippled [path/to/pipe/to]',
+  )
+  process.exit(1)
+}
+
+////////////////////////////////////////////////////////////////////////
+//  Get all necessary files from rippled
+////////////////////////////////////////////////////////////////////////
+const path = require('path')
+
+const fs = require('fs')
+function readFile(filename) {
+  try {
+    return fs.readFileSync(filename).toString('utf-8')
+  } catch (err) {
+    console.error(`Error reading file ${filename}:`, err.message)
+    process.exit(1)
+  }
+}
+
+const sfieldHeaderFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/SField.h'),
+)
+const sfieldMacroFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/detail/sfields.macro'),
+)
+const ledgerFormatsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/ledger_entries.macro',
+  ),
+)
+const terFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/TER.h'),
+)
+const transactionsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/transactions.macro',
+  ),
+)
+
+const capitalizationExceptions = {
+  NFTOKEN: 'NFToken',
+  URITOKEN: 'URIToken',
+  URI: 'URI',
+  UNL: 'UNL',
+  XCHAIN: 'XChain',
+  DID: 'DID',
+  ID: 'ID',
+  AMM: 'AMM',
+}
+
+// Translate from rippled string format to what the binary codecs expect
```

Comment:
There is duplicated effort across the client libraries to represent the Transaction Flags(`tf...`), Ledger-Specific-Flags (`lsf...`) and Account-Specific-Flags (`asf...`). xrpl.js has interfaces for easy inter-convertibility between numbers and flags representation, but xrpl-py has an ad-hoc implementation for transaction-flags only.

If we were to parse the flags values from rippled source files (similar to what is being accomplished in this file), it could reduce the development effort. 

Alternatively, this could be a great addition to the OpenAPI YAML specification project too.

Reply:
That's definitely a good extension of this script. I would say it's out of scope of this PR, though.